perception/lidar_apollo_segmentation_tvm/test/main.cpp

```diff
@@ -37,9 +39,34 @@ void test_segmentation(bool use_intensity_feature, bool use_constant_feature, bo
   const float objectness_thresh = 0.5f;
   const int32_t min_pts_num = 3;
   const float height_thresh = 0.5f;
+
+  std::string home = std::getenv("HOME");
+  fs::path data_path(home);
+  data_path /= "autoware_data";
+  fs::path apollo_data_path(data_path);
+  apollo_data_path /= "lidar_apollo_segmentation_tvm";
+  fs::path deploy_path(apollo_data_path);
+  deploy_path /= "models/baidu_cnn";
+
+  fs::path deploy_graph("deploy_graph.json");
+  fs::path deploy_lib("deploy_lib.so");
+  fs::path deploy_param("deploy_param.params");
+
+  fs::path deploy_graph_path = deploy_path / deploy_graph;
+  fs::path deploy_lib_path = deploy_path / deploy_lib;
+  fs::path deploy_param_path = deploy_path / deploy_param;
+
+  if (
+    !fs::exists(deploy_graph_path) || !fs::exists(deploy_lib_path) ||
+    !fs::exists(deploy_param_path)) {
+    printf("Model deploy files not found. Skip test.\n");
+    GTEST_SKIP();
```

Comment:
nit: would it get logged in a better way with `GTEST_SKIP() << "Model deploy files not found";` instead of printf? (https://google.github.io/googletest/advanced.html#skipping-test-execution)

Reply:
I tried yesterday and it doesn't output anything just skip. Same as it doesn't work in the sampling_based_planner 
https://github.com/autowarefoundation/autoware.universe/blob/7789be489dc437d2f2c2b62dba0ddbf27746ff09/planning/sampling_based_planner/sampler_common/test/test_transform.cpp#L49
![Screenshot from 2023-11-06 18-00-18](https://github.com/autowarefoundation/autoware.universe/assets/37497658/7e5482bb-4c4d-48e3-9cb5-fd963d628d68)

Or Is it only my local issue with gtest?

Reply:
I know `SUCCEED() << "msg";` doesn't print anything, I can see it being the same thing for the skip macro. The printf way looks good enough in the logs (https://github.com/autowarefoundation/autoware.universe/actions/runs/6783547137/job/18438043868?pr=5431#step:7:312):

```
1: [ RUN      ] lidar_apollo_segmentation_tvm.others
1: Model deploy files not found. Skip test.
1: Model deploy files not found. Skip test.
1: Model deploy files not found. Skip test.
1: Model deploy files not found. Skip test.
1: [  SKIPPED ] lidar_apollo_segmentation_tvm.others (0 ms)
```

Another request then: could this skip logic be moved inside of `TEST(lidar_apollo_segmentation_tvm, others)` below so that it only gets executed once instead of 4 times?

Reply:
Yes, I updated the code. It check for files only once and provide data_path to test function. 
Output looks like this:
![image](https://github.com/autowarefoundation/autoware.universe/assets/37497658/20f4adcd-241f-4c9a-9fa0-e3ab0c9ffdfc)